plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -425,6 +437,59 @@ public void testRegisterTableWithInvalidMetadataFileName()
         }
     }
 
+    @Test
+    public void testRegisterHadoopTableAndRead()
+            throws IOException
+    {
+        // create a temporary table to generate data file
+        String tempTableName = "temp_table_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tempTableName + " (id INT, name VARCHAR)");
+        assertUpdate("INSERT INTO " + tempTableName + " values(1, 'INDIA')", 1);
+        String dataFilePath = (String) computeScalar("SELECT \"$path\" FROM " + tempTableName);
+
+        // create hadoop table
+        String hadoopTableName = "hadoop_table_" + randomNameSuffix();
+        String hadoopTableLocation = metastoreDir.getPath() + "/" + hadoopTableName;
+        HadoopTables hadoopTables = new HadoopTables(newEmptyConfiguration());
+        Schema schema = new Schema(ImmutableList.of(
+                Types.NestedField.optional(1, "id", Types.IntegerType.get()),
+                Types.NestedField.optional(2, "name", Types.StringType.get())));
+        Table table = hadoopTables.create(
+                schema,
+                PartitionSpec.unpartitioned(),
+                SortOrder.unsorted(),
+                ImmutableMap.of(),
+                hadoopTableLocation);
+
+        // copy and append data file to hadoop table
+        String hadoopTableDataLocation = hadoopTableLocation + "/data";
+        Files.createDirectory(Path.of(hadoopTableDataLocation));
+        Files.copy(Path.of(dataFilePath), Path.of(hadoopTableDataLocation, "file-a.orc"));
+        DataFile fileA =
+                DataFiles.builder(PartitionSpec.unpartitioned())
+                        .withFormat(FileFormat.ORC)
+                        .withInputFile(localInput(new File(dataFilePath)))
+                        .withPath(hadoopTableDataLocation + "/file-a.orc")
+                        .withRecordCount(1)
+                        .build();
+        table.newFastAppend()
+                .appendFile(fileA)
+                .commit();
+
+        assertThat(getLatestMetadataLocation(fileSystem, hadoopTableLocation))
+                .endsWith("v2.metadata.json");
+
+        // Try registering hadoop table in trino and read the same
+        String registeredTableName = "registered_table_" + randomNameSuffix();
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(registeredTableName, hadoopTableLocation));
+        assertQuery("SELECT * FROM " + registeredTableName, "VALUES (1, 'INDIA')");
+        assertUpdate("INSERT INTO " + registeredTableName + " VALUES (2, 'POLAND')", 1);
+        assertQuery("SELECT * FROM " + registeredTableName, "VALUES (1, 'INDIA'), (2, 'POLAND')");
+
```

Comment:
```suggestion
        assertThat(Location.of(getLatestMetadataLocation(fileSystem, hadoopTableLocation)).fileName())
                .matches("00003-.*\\.metadata\\.json");
```

Reply:
I'm thinking it would be worth this way to emphasize the transition from  Hadoop style metadata version file names `vV.metadata.json`  towards `V.metadata.json` metadata file names.

Related Iceberg specification [section](https://iceberg.apache.org/spec/#file-system-tables)